packages/vue/docs/contributing/coding-guidelines.md

```diff
@@ -73,18 +73,157 @@ For example: `.gallery__item--selected`
 Keep up to 2 BEM levels (elements) at **most**.
 :::
 
+### CSS Custom Properties
 
-### CSS structure
+#### Rules and tips to create and use CSS Custom Properties / CSS Variables. 
+
+1. **DO NOT** use SASS VARIABLES.
+
+2. **Create variable** with default value like this:
+
+```css
+.sf-button {
+    padding: var(--button-padding, var(--spacer));
+}
+```
+:::tip 
+Create variables for all customizable properties like 
+`padding`, `margin`, `order`, `z-index`, `color`, `border`, `border-radius`, `background-color`. 
+
+Think of properties you would like to change it as a user and create for it a variable.
+
+
+Create size variables `var(--component-size, 25px)` in case when you have the same width and height value 
+
+```css
+.sf-button{
+  width: var(--component-width, var(--component-size, 266px));
+  height: var(--component-height, var(--component-size, 266px));
+}
+```
+:::
+
+3. Use **global variables** for spacers, font-sizes, colors. You can find all variables here `shared/styles/variables`
+
+:::tip
+SCSS functions and mixins you cand find here `shared/styles/_helpers.scss`
+
+All global variables are here `shared/styles/_variables.scss`
+:::
+
+4. **CSS Variables Naming convention**
+
+- variable name for BLOCK **var(--input-width);**
+- variable name for BLOCK elements **var(--input-label-padding);**
+```css
+.sf-input{
+  width: var(--input-width, 350px)
+  &__label{
+    padding: var(--input-label-padding, var(--spacer-small) var(--spacer))}
+  }
+```
+
+5. If you want to customize some component e.g. SfButton which is nested in molecule or organism you should use defined variable from SfButton to customize it.
+
+```css
+.sf-add-to-card {
+  &__add-button {
+    --button-padding: var(--spacer-medium);
+  }
+}
+```
+6.  Write variables always on the top of block before other properties.
+```css
+.sf-arrow{
+  --button-width: 20px;
+  --button-height: 20px;
+  outline: 0
+}
+```
+7. To create **font variable** use scss function 
+**font(--component-name, font-weight, font-size, line-height, font-family);**
+
+- Use it like this:
+
+```css
+  font: var(
+    --banner-title-font,
+    font(
+      --banner-title,
+      300,
+      var(--font-size-small),
+      1.6,
+      var(--body-font-family-secondary)
+    )
+  );
+``` 
+
+- If you want to customize:
+```css
+.sf-banner {
+    &--slim {
```

Comment:
@martaradziszewska please add declaration for &__title{...} to make it more transparent 👍